src/brain/gocdSlackFeeds/index.ts

```diff
@@ -128,43 +181,44 @@ const engineeringFeed = new DeployFeed({
   replyCallback: async (pipeline) => {
-    const hasFailedCanary =
-      pipeline.stage.name.includes('canary') &&
-      pipeline.stage.result.toLowerCase() === 'failed' &&
-      pipeline.stage.jobs
-        .find((job) => job.name === 'deploy-backend')
-        ?.result.toLowerCase() === 'failed';
-    if (!hasFailedCanary) return [];
+    const pauseCause = getPauseCause(pipeline);
+
+    if (pauseCause == null) return [];
     const [base, head] = await getBaseAndHeadCommit(pipeline);
     const authors = head ? await getAuthors('getsentry', base, head) : [];
-    // Get all users who have a slack account
-    const users = filterNulls(
-      await Promise.all(
-        authors.map((author) =>
-          getUser({ email: author.email, githubUser: author.login })
-        )
-      )
-    ).filter((user) => user.slackUser);
-    // Filter out duplicate users
-    const uniqueUsers = users.filter(
-      (user, index, self) =>
-        index === self.findIndex((u) => u.slackUser === user.slackUser)
-    );
+    // Get unique users from the authors
+    const uniqueUsers = await getUniqueUsers(authors);
+
     // Pick at most 10 users to cc
     const ccUsers = uniqueUsers.slice(0, 10);
     const ccString = ccUsers
       .map((user) => {
         return `<@${user.slackUser}>`;
       })
       .join(' ');
-    const gocdLogsLink = `https://deploy.getsentry.net/go/tab/build/detail/deploy-getsentry-backend-us/${pipeline.counter}/deploy-canary/${pipeline.stage.counter}/deploy-backend`;
-    const sentryReleaseLink = `https://sentry.sentry.io/releases/backend@${head}/?project=1`;
+
+    const failedJob = pipeline.stage.jobs.find(
+      (job) => job.result.toLowerCase() === 'failed'
+    );
+    if (!failedJob) {
+      // This should never happen, but if it does, we want to know about it
+      Sentry.captureException(
+        new Error('Failed to find failed job in failed pipeline')
+      );
+      return [];
+    }
+    const gocdLogsLink = `https://deploy.getsentry.net/go/tab/build/detail/${pipeline.name}/${pipeline.counter}/${pipeline.stage.name}/${pipeline.stage.counter}/${failedJob.name}`;
+    const sentryReleaseLink = pipeline.name.includes('s4s')
+      ? `https://sentry-st.sentry.io/releases/backend@${head}/?project=1513938`
+      : `https://sentry.sentry.io/releases/backend@${head}/?project=1`;
 
     const blocks = [
-      header(plaintext(':double_vertical_bar: Canary has been paused')),
+      header(
+        plaintext(`:double_vertical_bar: ${pipeline.name} has been paused`)
+      ),
       section(
-        markdown(`The deployment pipeline has been paused due to detected issues in canary. Here are the steps you should follow to address the situation:\n
-:mag_right: *Step 1: Review the Errors*\n Review the errors in the *<${gocdLogsLink}|Canary Logs>*.\n
+        markdown(`The deployment pipeline has been paused due to detected issues in ${pauseCause}. Here are the steps you should follow to address the situation:\n
```

Comment:
Does it make sense to say `The deployment pipeline has been paused due to detected issues in soak-time`? Is soak-time in s4s? I am missing context on soak time but guessing that other devs would too

Reply:
Soak time is the period following a deploy that we let the recently deployed changes sit and "soak" while we collect info about potential problems. It is basically post-deploy monitoring. I use the term "soak-time" as that is what is used in the pipelines and I wanted to keep naming consistent.

Reply:
With that said, I could add more/clearer information about soak time in the message

Reply:
Ah, it is a stage in the deployment pipeline entirely. Ok, that makes sense